### PR TITLE
Allow the AI to interact with sentries.

### PR DESCRIPTION
**Pull request by lbnesquik**
## About The Pull Request
- This also allow ghosts to check things out.

By the way, how the heck does manual override works? 

## Changelog
:cl:
expansion: Allow the AI to interact with turrets, and ghosts to check them out as well. No they can't use them.
/:cl:

### CODE REVIEW

File: code/game/objects/machinery/sentries.dm [hunk 1]
```diff
@@ -403,14 +403,16 @@
 		"burst_size" = burst_size,
 		"mini" = istype(src, /obj/machinery/marine_turret/mini)
 	)
+/obj/machinery/marine_turret/interact(mob/user)
+	ui_interact(user)
```
Comment (TiviPlus) [inline code/game/objects/machinery/sentries.dm:407]: there is a flag for this

**Review by TiviPlus — CHANGES_REQUESTED** [review]
.

File: code/game/objects/machinery/sentries.dm (hunk 1 above)
Comment (lbnesquik) [reply code/game/objects/machinery/sentries.dm:407]: I am not familiar with this flag

Comment (lbnesquik) [reply code/game/objects/machinery/sentries.dm]: Applied the changes

File: code/game/objects/machinery/sentries.dm [hunk 2]
```diff
@@ -410,7 +411,7 @@
 		return
 
 	var/mob/living/carbon/human/user = usr
-	if(!istype(user))
+	if(!istype(user) && !issilicon(user))
```
Comment (BraveMole) [inline code/game/objects/machinery/sentries.dm:414]: ew, no.

First you check what type is usr, then you cast.

And do you even need to cast anyway? when are you using user as a human or even as a mob?

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/sentries.dm (hunk 2 above)
Comment (lbnesquik) [reply code/game/objects/machinery/sentries.dm:414]: This stuff will get overwritten by the turret rework in like, two weeks, i don't think it needs to be as sturdy as regular code.

**Comment by lbnesquik** [comment]
Welllll, given the sentry rework is being test merged, i'll probably just wait until that is done

**Comment by SplinterGP** [comment]
> 
> 
> Welllll, given the sentry rework is being test merged, i'll probably just wait until that is done

do it, the sentries on the sentry rework uses a completely different typepath and this doenst affect them.

**Pull request closed without merging**